src/Pangraph/Internal/Hexml.hs

```diff
@@ -0,0 +1,11 @@
+module Pangraph.Internal.Hexml (
+hexmlParse
+) where
+
+import qualified Text.XML.Hexml as H
+import qualified Data.ByteString as BS
+
+hexmlParse :: BS.ByteString -> Maybe H.Node
+hexmlParse t = case H.parse t of
```

Comment:
And here you can use `either`:

`hexmlParse = either (const Nothing) Just . H.parse`